examples/proxy.rs

```diff
@@ -0,0 +1,47 @@
+#![deny(warnings)]
+extern crate http;
+extern crate hyper;
+extern crate pretty_env_logger;
+
+use http::HttpTryFrom;
+use hyper::{Client, Server};
+use hyper::service::service_fn;
+use hyper::rt::{self, Future};
+use std::net::SocketAddr;
+use std::sync::Arc;
+
+fn main() {
+    pretty_env_logger::init();
+
+    let in_addr = ([127, 0, 0, 1], 3001).into();
+    let out_addr: SocketAddr = ([127, 0, 0, 1], 3000).into();
+
+    let client_arc = Arc::new(Client::new());
```

Comment:
The `Client` is basically a `Arc` internally, so no need to double arc it.

Reply:
Ok, I was wondering about that.